src/main/java/uk/gov/hmcts/ccd/auditlog/aop/AuditAspect.java

```diff
@@ -0,0 +1,74 @@
+package uk.gov.hmcts.ccd.auditlog.aop;
+
+import org.apache.commons.lang.StringUtils;
+import org.aspectj.lang.JoinPoint;
+import org.aspectj.lang.ProceedingJoinPoint;
+import org.aspectj.lang.annotation.Around;
+import org.aspectj.lang.annotation.Aspect;
+import org.aspectj.lang.reflect.MethodSignature;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.springframework.context.expression.AnnotatedElementKey;
+import org.springframework.expression.EvaluationContext;
+import org.springframework.expression.spel.SpelEvaluationException;
+import org.springframework.stereotype.Component;
+import uk.gov.hmcts.ccd.auditlog.LogAudit;
+
+import java.lang.reflect.Method;
+import java.util.List;
+
+@Aspect
+@Component
+public class AuditAspect {
```

Comment:
since we're using AOP I'll request some thorough performance testing on this ticket. Will add a comment on the ticket

Reply:
sure, but that's the cleanest option we came across. Any other design suggestions always welcome.

Reply:
I think AOP is a good choice. We just need to evaluate the performance of the solution.